### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by mhubii**
Refers to #149

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +35,12 @@ vcs import src < src/repos.yaml
 rosdep install --from-paths src --ignore-src -r -y
 colcon build --symlink-install
 ```
+Copy the Dockerfile and the container scripts to the lbr-stack directory. Build and start the container.
```
Comment (mhubii) [inline README.md]: consider moving this into demos, but neat in readme, too

File: lbr_humble_docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+FROM ros:humble-ros-base-jammy
+
+SHELL ["/bin/bash", "-c"]
+
+RUN apt update -y 
+RUN apt install vim python-is-python3 python3-pip python3-colcon-common-extensions -y
```
Comment (mhubii) [inline lbr_humble_docker/Dockerfile]: check all dependencies, maybe install `ros-dev-tools` instead here?

**Review by mhubii — COMMENTED** [review]
looks good overall, re-iterate on file placement

**Comment by mhubii** [comment]
hi @Nicolai-98 , thank you again for contributing this. I now have time to review properly and would like to merge into humble.

do you think it makes sense to put the shell scripts / the dockerfile into the root folder, since they are no ros package. Also, in terms of build instructions, does it make sense to separate docker instructions? Thank you for the input and the help!

**Comment by Nicolai-98** [comment]
Hi! I'm always happy to be able to contribute. Ideally you want the docker related files in the lbr-stack directory that is created using the quick start instructions from the ReadMe: `mkdir -p lbr-stack/src && cd lbr-stack`. This directory does not exist yet when cloning the repository. This is why I suggested copying them after following the quick start by using: 
```shell
cp -r src/lbr_fri_ros2_stack/lbr_humble_docker/* .
chmod +x container_build.sh 
sudo ./container_build.sh
```
To answer the first question: The files can also be put into the root directory as long as they end up in the correct directory after building the project. This allows an easy workflow where you only have to use the container_build script once and then you can start a container whenever you need it using the container_start script.
I am not 100% sure what you mean by separating docker instructions from build instructions. Could you clarify?

**Comment by mhubii** [comment]
okay got it! By that I mean just running with / without Docker.

Hm okay, would it be useful to turn these shell scripts into entry points a la

```shell
ros2 run lbr_docker build_docker
```

that would justify a package, right? Or is that rather pointless. Can `setup.py` turn shell scripts into entry points? But then that's also strange because one would have to install the package prior to building a container.

**Comment by mhubii** [comment]
okay I tested it and it works, got to double check on real robot. But I would say lets merge this PR ASAP and go from here. Thank you again @Nicolai-98 !

**Comment by Nicolai-98** [comment]
Its my pleasure. I think the current setup with the changes you made is perfectly fine. I don't think turning the shell scripts into entry points adds any extra value.

File: lbr_humble_docker/Dockerfile (hunk 2 above)
Comment (Nicolai-98) [reply lbr_humble_docker/Dockerfile]: Installing ros-dev-tools instead should work but I have not tested it yet. I could test it over the weekend.

**Review by Nicolai-98 — APPROVED** [review]
Looks good and tidy!

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+# create workspace
+RUN mkdir -p home/ros2_ws/src
+WORKDIR /home/ros2_ws/
+
+COPY ./src /home/ros2_ws/src
+RUN rosdep install -i --from-paths src --rosdistro ${ROS_DISTRO} -y 
+
+# "--symlink-install" allows the code in the locally mounted volume ./src to be adjusted without rebuilding
+RUN source /opt/ros/${ROS_DISTRO}/setup.bash && \
```
Comment (mhubii) [inline docker/Dockerfile:18]: okay so I cleaned some dependencies, as these are mostly in the base images already.

One more question before merge @Nicolai-98: I don't use Docker a lot, so please let me know if you need scipy / numpy / vim / python3 is python etc installed, so it is more convenient for your use.

There will likely be a couple changes moving forward but this is a great first step.

Comment (mhubii) [reply docker/Dockerfile:18]: Sorry I went ahead and merged this anyways already. Thank you again! You are now listed among contributors. 

Please feel free to create new PRs against `humble` if you see fit.

**Pull request merged**